lib/debugger/debugger.dart

```diff
@@ -1273,6 +1290,69 @@ class VariablesView implements CoreElementView {
   }
 }
 
+class VariablesChildProvider extends ChildProvider<BoundVariable> {
+  VariablesChildProvider(this.debuggerState);
+
+  final DebuggerState debuggerState;
+
+  @override
+  bool hasChildren(BoundVariable item) {
+    final dynamic value = item.value;
+    return value is InstanceRef && value.valueAsString == null;
+  }
+
+  @override
+  Future<List<BoundVariable>> getChildren(BoundVariable item) async {
+    final dynamic value = item.value;
+    if (value is! InstanceRef) {
+      return [];
+    }
+
+    final InstanceRef instanceRef = value;
+    final dynamic result = await debuggerState.getInstance(instanceRef);
+    if (result is! Instance) {
+      return [];
+    }
+
+    // TODO: how to test?
```

Comment:
have an intermediate abstraction that is not browser specific made the similar code for the inspector a lot easier to test and will help with a hummingbird renderer int he future.